File: tdd-01/code/src/com/company/model/Car.java
package com.company.model;

/**
 * @author Olivier PEREZ
 */
public class Car {

	private Engine engine;
	
	private Alarm alarm;

	public Car (final Engine engine) {
		this.engine = engine;
	}

	public void setAlarm(final Alarm alarm) {
		this.alarm = alarm;
	}

	public String start() {
		return engine.start();
	}
}

File: tdd-01/code/objectivec/TDD-Car-01/model/Car.m
//
//  Car.m
//  TDD-Car-01
//
//  Created by AWL_02 on 16/06/2014.
//  Copyright (c) 2014 Company. All rights reserved.
//

#import "Car.h"
#import "Engine.h"

@implementation Car

- (id)initWithEngine:(Engine *)engine_ {

    if ( self = [super init] ) {
        self->engine = engine_;
        return self;

    } else

        return nil;

}

-(NSString*)start {
    return [engine start];
}

@end
